python/ray/data/datasource/image_datasource.py

```diff
@@ -132,35 +60,28 @@ def create_reader(
         _check_import(self, module="PIL", package="Pillow")
         _check_import(self, module="pandas", package="pandas")
 
-        # We call `_resolve_paths_and_filesystem` so that the dataset root is formatted
-        # in the same way as the paths passed to `_get_class_from_path`.
-        paths, filesystem = _resolve_paths_and_filesystem([root])
-        root = paths[0]
+        return _ImageDatasourceReader(self, size=size, mode=mode, **kwargs)
 
-        return _ImageFolderDatasourceReader(
-            delegate=self,
-            paths=paths,
-            filesystem=filesystem,
-            partition_filter=FileExtensionFilter(file_extensions=IMAGE_EXTENSIONS),
-            root=root,
-            size=size,
-            mode=mode,
-        )
+    def _convert_block_to_tabular_block(
+        self, block: "pyarrow.Table", column_name: Optional[str] = None
+    ) -> "pyarrow.Table":
+        # The input block has one column named `TENSOR_COLUMN_NAME`. We don't want to
+        # leak `TENSOR_COLUMN_NAME`, so we rename the column to a more human-readable
+        # name.
+        return block.rename_columns([self._COLUMN_NAME])
 
     def _read_file(
         self,
         f: "pyarrow.NativeFile",
         path: str,
-        root: str,
         size: Optional[Tuple[int, int]],
         mode: Optional[str],
-    ) -> Block:
-        import pandas as pd
+    ) -> "pyarrow.Table":
```

Comment:
Should be `Block` if we end up using `DelegatingBlockBuilder`.
```suggestion
    ) -> Block:
```